src/Bicep.Core/Analyzers/Linter/Rules/BCPL1060.cs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Parsing;
+using Bicep.Core.PrettyPrint;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1060 : LinterRule
+    {
+        public BCPL1060() : base(
+            code: "BCPL1060",
+            ruleName: "Dynamic variable used concat",
+            description: "Dynamic variable should not use concat - string interpolation should be used.",
+            docUri: "https://bicep/linter/rules/BCPL1060") // TODO: setup up doc pages
+        { }
+
+        private CodeReplacement GetCodeReplacement(TextSpan span)
+            => new CodeReplacement(span, "BCPL1060 - this is the new code");
+
+        override public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            var funcSymbols = model.Root.Declarations.OfType<VariableSymbol>()
+                                    .Select(sym => sym.Value)
+                                    .OfType<FunctionCallSyntax>();
+
+            foreach (var func in funcSymbols.Where(f => f.Name.IdentifierName.Equals("concat", System.StringComparison.OrdinalIgnoreCase)))
```

Comment:
>Equals [](http://example.com/codeflow?start=78&length=6)

Technically multiple namespaces can contain a function called `concat`. I think the best practice is to check this against the symbol table using both the namespace and function name. (And probably make it easy for linter authors as well.)

Reply:
This would apply anywhere where we are string comparing function names.

Reply:
This should be fixed in all rules.